src/rust/engine/process_execution/src/docker.rs

```diff
@@ -203,9 +207,13 @@ impl super::CommandRunner for CommandRunner {
   async fn run(
     &self,
     context: Context,
-    _workunit: &mut RunningWorkunit,
+    workunit: &mut RunningWorkunit,
     req: Process,
   ) -> Result<FallibleProcessResultWithPlatform, ProcessError> {
+    if req.docker_image.is_none() {
```

Comment:
I disagree with `docker::CommandRunner` needing to own any other command runner. In terms of encapsulation, `docker::CommandRunnner` fits more naturally in my view as a leaf command runner, and not some combination of leaf and combinator. The combinator to write here is a `DockerSwitchCommandRunner` which does this very conditional and selects one of the two command runners it owns.

Reply:
This seems like overkill? We already have decent encapsulation via using `Box<dyn crate::CommandRunner>` (whereas before I was using `local::CommandRunner`, bad).

Defining another CommandRunner involves a lot of boilerplate.

Using this current approach has seemed to work fine for Nailgun?

Reply:
> Using this current approach has seemed to work fine for Nailgun?

Nailgun actually was accessing fields on `local::CommandRunner` and thus was written in such a way that it had to know about the internals of `local::CommandRunner`. There is no such need with `docker::CommandRunner`.